ui/nuxeo-document-comments/nuxeo-document-comment-thread.js

```diff
@@ -0,0 +1,293 @@
+          <span class="more-content no-selection pointer smaller" on-tap="_loadMore"
+            >[[_computeTextLabel(level, 'loadAll', total, i18n)]]</span
+          >
+        </template>
+        <template id="commentList" is="dom-repeat" items="[[comments]]" as="comment">
+          <nuxeo-document-comment comment="{{comment}}" level="[[level]]"></nuxeo-document-comment>
+        </template>
+
+        <template is="dom-if" if="[[_allowReplies(level)]]">
+          <div class="input-area">
+            <paper-textarea
+              id="inputContainer"
+              placeholder="[[_computeTextLabel(level, 'writePlaceholder', null, i18n)]]"
+              value="{{text}}"
+              max-rows="[[_computeMaxRows()]]"
+              no-label-float
+              on-keydown="_checkForEnter"
+            >
+            </paper-textarea>
+            <template is="dom-if" if="[[!_isBlank(text)]]">
+              <iron-icon
+                id="submit"
+                name="submit"
+                class="main-option opaque"
+                icon="check"
+                on-tap="_submitComment"
+              ></iron-icon>
+              <nuxeo-tooltip for="submit">[[i18n('comments.submit.tooltip')]]</nuxeo-tooltip>
+              <iron-icon name="clear" class="main-option opaque" icon="clear" on-tap="_clearInput"></iron-icon>
+            </template>
+          </div>
+        </template>
+      `;
+    }
+
+    static get is() {
+      return 'nuxeo-document-comment-thread';
+    }
+
+    static get properties() {
+      return {
+        uid: {
+          type: String,
+          observer: '_refresh',
+        },
+
+        comments: {
+          type: Array,
+          value() {
+            return [];
+          },
+        },
+
+        level: {
+          type: Number,
+          value: 1,
+        },
+
+        pageSize: {
+          type: Number,
+          readOnly: true,
+          value: 10,
+        },
+
+        allCommentsLoaded: {
+          type: Boolean,
+          readOnly: true,
+          value: false,
+        },
+
+        total: {
+          type: Number,
+          readOnly: true,
+          value: 0,
+        },
+      };
+    }
+
+    connectedCallback() {
+      super.connectedCallback();
+      this.addEventListener('delete-comment', this._handleDeleteEvent);
+      this.addEventListener('edit-comment', this._handleEditEvent);
+      this.addEventListener('comments-changed', this._handleCommentsChange);
+    }
+
+    disconnectedCallback() {
+      this.removeEventListener('delete-comment', this._handleDeleteEvent);
+      this.removeEventListener('edit-comment', this._handleEditEvent);
+      this.removeEventListener('comments-changed', this._handleCommentsChange);
+      super.disconnectedCallback();
+    }
+
+    focusInput() {
+      this.$$('#inputContainer').focus();
+    }
+
+    _checkForEnter(e) {
+      if (e.keyCode === 13 && e.ctrlKey) {
+        if (!this._isBlank(this.text)) {
```

Comment:
Merge ifs.

Reply:
Regarding this, what do you think about extracting the input container (logic and HTML) to an inner element since we are replicating some code on both elements (`nuxeo-document-comment` and ` nuxeo-document-comment-thread`)?

Reply:
I agree that having such an element could be useful. This means that `--nuxeo-comment-line-height` and `--nuxeo-comment-max-height` would be part of that element.